x-pack/plugins/fleet/server/services/package_policies/experimental_datastream_features.ts

```diff
@@ -74,6 +75,31 @@ export async function handleExperimentalDatastreamFeatureOptIn({
       // @ts-expect-error - TODO: Remove when ES client typings include support for synthetic source
       body,
     });
+
+    if (featureMapEntry.features.TSDB) {
+      const indexTemplateRes = await esClient.indices.getIndexTemplate({
+        name: featureMapEntry.data_stream,
+      });
+      const indexTemplate = indexTemplateRes.index_templates[0].index_template;
+
+      const indexTemplateBody = {
+        ...indexTemplate,
+        template: {
+          ...(indexTemplate.template ?? {}),
+          settings: {
+            ...(indexTemplate.template?.settings ?? {}),
+            index: {
+              mode: 'time_series',
+            },
+          },
+        },
+      };
+
+      await esClient.indices.putIndexTemplate({
```

Comment:
It is not possible to put that settings in the `@package` component template instead of the index template?

Reply:
No, it is not possible: https://github.com/elastic/elasticsearch/issues/74660#issuecomment-1309940237